llmfoundry/data/finetuning/tasks.py

```diff
@@ -371,10 +373,19 @@ def __init__(self,
         )
 
         self.tokenizer = tokenizer
+        self.max_seq_len = max_seq_len
 
     # How to process a sample
     def __getitem__(self, idx: int) -> Dict[str, Any]:
         sample = super().__getitem__(idx)
+        if 'input_ids' in sample:
+            # already tokenized data
+            sample['input_ids'] = np.frombuffer(
```

Comment:
were the tolist calls necessary here? I notice that the equivalent code in `text_data.py` doesn't have the tolist

Reply:
Prior to this change, Line 389 returns a python list (not a numpy array or torch.Tensor). So in order to make it return the same type as before, I convert it to a list. 

The code in text_data.py returns a tensor by calling `torch.from_numpy()`, so that it doesn't need the tolist().

Reply:
ah, makes sense, thanks!